### PR TITLE
ARROW-6891: [Rust] [Parquet] utf8 support for arrow reader.

### PR DESCRIPTION
**Pull request by liurenjie1024**
UTF8 support for arrow reader.

### CODE REVIEW

**Comment by liurenjie1024** [comment]
@sunchao @andygrove @nevi-me @paddyhoran Please take a look when your are available.

File: rust/parquet/src/arrow/array_reader.rs [hunk 1]
```diff
@@ -653,11 +830,14 @@ impl<'a> ArrayReaderBuilder {
...
             )),
-            PhysicalType::BYTE_ARRAY => Ok(Box::new(PrimitiveArrayReader::<
-                ByteArrayType,
-            >::new(
-                page_iterator, column_desc
-            )?)),
+            PhysicalType::BYTE_ARRAY
+                if cur_type.get_basic_info().logical_type() == LogicalType::UTF8 =>
+            {
+                Ok(Box::new(ComplexObjectArrayReader::<
```
Comment (nevi-me) [inline rust/parquet/src/arrow/array_reader.rs]: Out of curiousity, would the `ComplexObjectArrayReader` allow us to implement `arrow::BinaryArray` and `arrow::FixedSizeBinaryArray` in future?

**Review by nevi-me — APPROVED** [review]
LGTM

File: rust/parquet/src/arrow/array_reader.rs (hunk 1 above)
Comment (liurenjie1024) [reply rust/parquet/src/arrow/array_reader.rs]: Yes, we just need converter for those types.

**Review by andygrove — APPROVED** [review]
Thanks @liurenjie1024

**Pull request closed without merging**